azurerm/internal/provider/provider.go

```diff
@@ -312,6 +319,7 @@ func providerConfigure(p *schema.Provider) schema.ConfigureFunc {
 			TerraformVersion:            terraformVersion,
 			PartnerId:                   d.Get("partner_id").(string),
 			DisableCorrelationRequestID: d.Get("disable_correlation_request_id").(bool),
+			CustomCorrelationRequestID:  d.Get("custom_correlation_request_id").(string),
```

Comment:
since we're not exposing this in the Provider block, we can pull from the env var here directly, e.g.

```suggestion
			CustomCorrelationRequestID:  os.GetEnv("ARM_CORRELATION_REQUEST_ID"),
```